plugins/transport-reactor-netty4/src/main/java/org/opensearch/http/reactor/netty4/ReactorNetty4NonStreamingHttpChannel.java

```diff
@@ -46,7 +46,13 @@ public boolean isOpen() {
 
     @Override
     public void close() {
-        request.withConnection(connection -> connection.channel().close());
+        request.withConnection(connection -> {
```

Comment:
Run into the case when channel is disposed and Reactor Netty does not use `closeFuture`, always creates a new future instead (for reference, `DisposedChannel`)